Cargo.toml

```diff
@@ -45,7 +45,9 @@ cfg-if = "1.0.0"
 clap = "4.5.9"
 clap-cargo = "0.14.1"
 itertools = "0.13.0"
-lazy_static = "1.5.0"
+once_cell = { version = "1.20.2", default-features = false, features = [
+  "alloc",
```

Comment:
hmm is alloc required here? this forces there to be an allocator pulled in always. packages without them would fail to compile

Reply:
To be addressed in separate issue: we should add some samples to the top level test that explicitly test this no alloc available case. The current examples all pull in `wdk-alloc` but samples should work without it still

Reply:
I think as it stands, this pr breaks the pure no-std case

Reply:
@leon-xd  this was marked as resolved, but i believe its still the case?